app/code/core/Mage/GoogleAnalytics/Block/Ga.php

```diff
@@ -72,11 +72,49 @@ protected function _getPageTrackingCode($accountId)
     {
         /** @var Mage_GoogleAnalytics_Helper_Data $helper */
         $helper = $this->helper('googleanalytics');
-        if ($helper->isUseUniversalAnalytics()) {
+        if ($helper->isUseAnalytics4()) {
+            return $this->_getPageTrackingCodeAnalytics4($accountId);
+        } elseif ($helper->isUseUniversalAnalytics()) {
             return $this->_getPageTrackingCodeUniversal($accountId);
+        } else {
+            return $this->_getPageTrackingCodeAnalytics($accountId);
+        }
+    }
+
+    /**
+     * Render regular page tracking javascript code
+     *
+     * @param string $accountId
+     * @return string
+     */
+    protected function _getPageTrackingCodeAnalytics4($accountId)
+    {
+        $trackingCode = "
+gtag('js', new Date());
+";
+        if(!$this->helper('googleanalytics')->isDebugModeEnabled()){
+            $trackingCode.= "
+gtag('config', '{$this->jsQuoteEscape($accountId)}');
+";
+        }else{
+            $trackingCode.= "
+gtag('config', '{$this->jsQuoteEscape($accountId)}', { 'debug_mode': true });
+";
+        }
+
+        //add user_id
+        if ($this->helper('googleanalytics')->isUserIdEnabled() && Mage::getSingleton('customer/session')->isLoggedIn()) {
+            $customer = Mage::getSingleton('customer/session')->getCustomer();
+            $trackingCode.= "
+gtag('set', 'user_id', '{$customer->getId()}');
+";
```

Comment:
```suggestion
            $trackingCode.= "gtag('set', 'user_id', '{$customer->getId()}');";
```

Reply:
new line follow same style in _getPageTrackingCodeUniversal, i think new line is for bettere readability in frontend
i know it's awful, we can use <<<EOF or start with "\ngtag(....

Reply:
mmm persomally I much rather like a readable php than a readable generated html ;-)